### PR TITLE
feat allow visibility files by date

### PR DESCRIPTION
**Pull request by johnvente**
## Description
This PR enables users to specify the dates when files/folders will be visible in the LMS. Users can select a range of dates, from a start date to an end date, during which the file/folder will be visible.

**DEMO**

https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/0de9b647-7ca3-40fc-a15d-7b4e6fb27558

### How to test it

- Check this branch`jv/feat-adding-metadata-to-files` in the XBlock files manager

1. use node 18
 ```
nvm use
```

2. install dependencies
```
yarn install 
```
3 run to see any change in the react code 
```
yarn dev 

```
-----
### Setting a date
1. Select a file/folder
2. Right click on "Set file visibility with date" button
set-vibistily-date

3. Chose the range of dates
range-dates

4. Click on "Save" button

### Removing a date
1. Select a file/folder
2.  Right click on  "Remove date visibility" button
remove-dates

3. Click on "Save" button

**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Fixup commits are squashed away
- [ ] Unit tests added/updated
- [ ] Manual testing instructions provided
- [ ] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

### CODE REVIEW

File: react-app/components/FileManager/index.jsx
Comment (BryanttV) [inline react-app/components/FileManager/index.jsx:1]: Hi @johnvente, the implementation works great, I just have a few comments:

1. In the handler response the files are returned anyway, aren't we going to remove them from there too?
2. With the date picker it is not possible to configure by days, is there a way to do it including a time? or is not required this option?
3. We should update the instructions and documentation including the new date functionality.
4. Is it possible to set dates also for folders? If it is not possible, I think that the option should not appear when selecting a folder.

Additionally,

1. When I try to download several files (including folders) I get an error message, but it still generates the zip to download.
image

**Review by BryanttV — CHANGES_REQUESTED** [review]

File: react-app/components/FileManager/constants.js
[the remark this answers is not in the document]
Comment (johnvente) [reply react-app/components/FileManager/constants.js:1]: Good idean, thanks

File: react-app/components/FileManager/index.jsx
Comment (johnvente) [reply react-app/components/FileManager/index.jsx:1]: Hi there @BryanttV 
1. I suggested that to feliple but he said that wouldn't be neccessary
2. It's not required the time, at least for now
3. I will add the instructions and the documentation for this
4. It's possible to do it for folders but it seems like a backend error isn't saving the metadata of those one

**Comment by johnvente** [comment]
Hi @BryanttV I've solved the problem with the folders and I added the instructions could you check again please?

**Comment by BryanttV** [comment]
Hi @johnvente, thanks. LGTM!

**Review by BryanttV — APPROVED** [review]

**Comment by johnvente** [comment]
> Just one thing, could we update the cover letter with the new changes?

Thanks, done!

**Pull request merged**